contrib/dimfeld/httptreemux.v5/httptreemux_test.go

```diff
@@ -238,18 +238,777 @@ func TestNamingSchema(t *testing.T) {
 	namingschematest.NewHTTPServerTest(genSpans, "http.router")(t)
 }
 
-func TestTrailingSlashRoutes(t *testing.T) {
```

Comment:
While tests are highly appreciated, it would be preferred to keep this PR as short / simple as possible to make it easier to review and maintain.

Keeping the existing `TestTrailingSlashRoutes` test as it is with a new case covering this bug should be enough.

Reply:
I can definitely look into reducing some of the redundant test cases, but I do think at least some additional test cases for the 4 router redirect modes is warranted, i.e. (from `dimfeld/httptreemux/v5/router.go`):

```golang
package httptreemux

const (
	Redirect301 RedirectBehavior = iota // Return 301 Moved Permanently
	Redirect307                         // Return 307 HTTP/1.1 Temporary Redirect
	Redirect308                         // Return a 308 RFC7538 Permanent Redirect
	UseHandler                          // Just call the handler function
)
```

Reply:
The test cases were reduced down to three scenarios per redirect mode:
1. path not found
2. path with parameterized URL _excluding_ a trailing slash
3. path with parameterized URL _including_ a trailing slash